app/components/NetworkDrawer/NetworkDrawer.tsx

```diff
@@ -0,0 +1,146 @@
+import { WhaleApiClient } from '@defichain/whale-api-client'
+import { MaterialIcons } from '@expo/vector-icons'
+import React, { useEffect, useRef, useState } from 'react'
+import { ActivityIndicator, Animated, Linking, TouchableOpacity, View } from 'react-native'
+import { useDispatch, useSelector } from 'react-redux'
+import tailwind from 'tailwind-rn'
+import { Text } from '..'
+import { Logging } from '../../api/logging'
+import { PrimaryColor, PrimaryColorStyle } from '../../constants/Theme'
+import { useWhaleApiClient } from '../../contexts/WhaleContext'
+import { RootState } from '../../store'
+import { firstTransactionSelector, networkDrawer, Transaction } from '../../store/networkDrawer'
+import { translate } from '../../translations'
+
+async function handlePress (txid: string): Promise<void> {
+  // TODO(thedoublejay) explorer URL
+  const url = `https://explorer.defichain.io/#/DFI/mainnet/tx/${txid}`
+  const supported = await Linking.canOpenURL(url)
+  if (supported) {
+    await Linking.openURL(url)
+  }
+}
+
+async function broadcastTransaction (tx: Transaction, client: WhaleApiClient, setError: (e: Error) => void): Promise<string | undefined> {
+  let retries = 0
+  try {
+    return await client.transactions.send({ hex: tx.signed.toHex() })
+  } catch (e) {
+    retries++
+    Logging.error(e)
+    if (retries < 1) {
+      return await broadcastTransaction(tx, client, setError)
+    }
+    setError(e)
+  }
+}
+
+/**
+ * @description - Global component to be used for async calls, network errors etc. This component is positioned above the bottom tab.
+ * Need to get the height of bottom tab via `useBottomTabBarHeight()` hook to be called on screen.
+ * */
+export function NetworkDrawer (): JSX.Element {
+  const { height, err: e } = useSelector((state: RootState) => state.networkDrawer)
+  const transaction = useSelector((state: RootState) => firstTransactionSelector(state.networkDrawer))
+  const [tx, setTx] = useState<Transaction>(transaction)
+  const dispatch = useDispatch()
+  const client = useWhaleApiClient()
+  const [err, setError] = useState<Error | undefined>(e)
+  const slideAnim = useRef(new Animated.Value(0)).current
+  Animated.timing(slideAnim, { toValue: height, duration: 300, useNativeDriver: false }).start()
+
+  useEffect(() => {
+    if (transaction !== undefined) {
+      broadcastTransaction(transaction, client, setError).then(() => {
+        setTx({
+          ...transaction,
+          title: translate('screens/NetworkDrawer', 'Transaction complete'),
+          broadcasted: true
+        })
+        dispatch(networkDrawer.actions.popTransaction())
+      }).catch(() => {
+      })
+    }
+  }, [transaction])
+
```

Comment:
```suggestion
  if (transaction === undefined) return null
```
By doing this we can remove all the `setHeight()` right?

Reply:
Nope, setHeight is required to have exact measurements of bottom drawer. It's unrelated to transaction but it's merely a UI state. Height varies per device so we need to set it to achieve the style of drawer